documentation/src/main/asciidoc/topical/wildfly/Wildfly.adoc

```diff
@@ -0,0 +1,54 @@
+= Updating Hibernate ORM within WildFly
+
+The http://wildfly.org/[WildFly application server] comes with Hibernate as the default JPA provider out of the box.
+This means that you don't need to package Hibernate ORM with the applications you deploy on WildFly,
+instead the application server will automatically enable Hibernate support if it detects that your application works with JPA/Hibernate.
+
+There may be times though where a newer version of Hibernate ORM is available than the one coming with a given WildFly release.
+For that case the Hibernate ORM project provides a ZIP file containing the required modules to update a WildFly installation to newer versions.
+The module ZIP file is available from SourceForge and BinTray, alike the ZIP/TAR.GZ distributions (TODO: Add link).
+As of Hibernate ORM 5.1.1, a ZIP file for upgrading WildFly 10 to that Hibernate version (from 5.0.x) is provided.
+The general goal is to provide a ZIP file for bringing new versions of Hibernate ORM to the WildFly version current at the time of the Hibernate release.
+
+Once downloaded, extract the contents of the ZIP file into the _modules_ directory of your WildFly installation
+(shut down the application server before, should it be running).
+Note that the Hibernate ORM modules coming with WildFly will remain untouched,
+i.e. you can switch between the original version and the new version from the ZIP file as needed as a matter of configuration.
+
+The module system of the application server uses what's called "slots" to differentiate between several versions of one module.
+By default, slot "main" of the Hibernate ORM modules will be used to provide JPA support for given deployments (representing the Hibernate version coming with WildFly itself).
+In order to use another slot, specify the following two property in the _persistence.xml_ file of your application:
```

Comment:
let's talk about "module ids" rather than slots (as for the reason above, but also clearer since the properties you're referring to require the full module name)